redwood/src/lib.rs

```diff
@@ -195,11 +209,35 @@ pub fn decrypt(
 #[cfg(test)]
 mod tests {
     use super::*;
+    use sequoia_openpgp::Cert;
     use tempfile::NamedTempFile;
 
     const PASSPHRASE: &str = "correcthorsebatterystaple";
     const SECRET_MESSAGE: &str = "Rust is great 🦀";
 
+    // Purposefully weak key that should be rejected; this is 1024-bit RSA
+    const BAD_KEY: &str = "-----BEGIN PGP PUBLIC KEY BLOCK-----
+        mI0EZN1mPgEEANxv1ymcSsTjSvx/itkHBUCuf/LCSiU+T/VhUPT+pSxO4nyE4aLF
+        SAawGnxwGLPg2nO4txX93eZRX+PHDa5/Uk105RfAqFafC1UDFBNu+dA5CPxJx7BT
+        2yTSdpJh1a/9ur0HWrad9PV6eYWirPo2vhqaXiqhYkLcSHVwmFfPNMLJABEBAAG0
+        LmRlbGxzYmVyZyAoRVhBTVBMRV9LRVlfT05MWSkgPGZvb0BleGFtcGxlLm9yZz6I
+        zgQTAQoAOBYhBFbO0Jy+SeQ+Qtc9NeUVm4kgzl8xBQJk3WY+AhsDBQsJCAcCBhUK
+        CQgLAgQWAgMBAh4BAheAAAoJEOUVm4kgzl8x2OoD/ROBhvNs1J48Ogbr8bgx/zCk
+        yIXteBEvzJe7dNkHY0fG/bGuHIrAiGH0LZt2Y+xkWaYF6laxjLmZbkw+MA9HAxnH
+        EKgXGFj59+trxL/bRhKIzHj96yxCRGCls+YzmVU3okK/UhYJ61ze38OX6QL4jrP9
+        ltAlaGlVjh9RPpGM4HF6uI0EZN1mPgEEAOhN5IcYtSQgeEvmAjeErrJf10P/9jdh
+        Xn3OpG0qLxiDba60BSw9OIGSfDP/zT+Q2XBw3Sla2p2Moe567ospeFpuJsM2pubO
+        O2IEt3+7CnzIM860P4MSGK24etJvcRxvR4m1m2Cr86xC4vfVltyEp6ON42Yb1i+C
+        R2q2qxmwQ8e7ABEBAAGItgQYAQoAIBYhBFbO0Jy+SeQ+Qtc9NeUVm4kgzl8xBQJk
+        3WY+AhsMAAoJEOUVm4kgzl8xdQYD/3ShTd2KM0XS25gMzfTQ8NtBS+jBvR0/+Eej
+        0BPhAf57pvNs8Y/uxUPO6l1+KrQek8XEnvGAkNBMD0wd2xPoQzcQqnf0iMcPhAY2
+        wCYRRONffypz3PkTM/JsFH08i2/UsntkgCLbgCEX/0Od1OFJT8VOSIOZe8EVjlGz
+        UUAkCN7u
+        =9bsJ
+        -----END PGP PUBLIC KEY BLOCK-----";
+    const BAD_KEY_FINGERPRINT: &str =
+        "56CED09CBE49E43E42D73D35E5159B8920CE5F31";
+
```

Comment:
This seems...kinda ugly, so I'm open to other ways of doing it, but I wasn't sure the best way to include an unsupported key, since the CertBuilder doesn't have an option for unsupported CipherSuites and I got a bit into the weeds looking at how to generate certs that we don't actually want to support :)